packages/draw/src/animation.ts

```diff
@@ -0,0 +1,123 @@
+type HexColor = string;
+type AnimateableProperty = number | HexColor;
+type AnimationState<T extends AnimateableProperty> = {
+  initial: T;
+  current: T;
+  desired: T;
+};
+
+export type Easing = 'linear' | 'easein' | 'flip' | 'easeout';
+
+// from: https://stackoverflow.com/questions/5623838/rgb-to-hex-and-hex-to-rgb
+function hexToRgb(hex: string): { r: number; g: number; b: number } {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+  return result
+    ? {
+        r: parseInt(result[1], 16),
+        g: parseInt(result[2], 16),
+        b: parseInt(result[3], 16),
+      }
+    : { r: 0, g: 0, b: 0 };
+}
+
+function componentToHex(c: number) {
+  const hex = Math.trunc(c).toString(16);
+  return hex.length == 1 ? '0' + hex : hex;
+}
+
+function rgbToHex(r: number, g: number, b: number) {
+  return '#' + componentToHex(r) + componentToHex(g) + componentToHex(b);
+}
+
+function linear(per: number) {
+  return per;
+}
+
+function easeIn(per: number) {
+  return per * per;
+}
+
+function flip(per: number) {
+  return 1 - per;
+}
+
+function easeOut(per: number) {
+  return flip(flip(per) ** 2);
+}
+
+export class Animation<T extends AnimateableProperty> {
+  public constructor({
+    from,
+    to,
+    easing,
+    duration,
+  }: {
+    from: T;
+    to: T;
+    easing: Easing;
+    duration: number;
+  }) {
+    this.state = {
+      initial: from,
+      current: from,
+      desired: to,
+    };
+    this.easing = easing;
+    this.duration = duration;
+    this.startTime = new Date().getTime();
+  }
+
+  public getValue(): T {
+    const currTime = new Date().getTime();
+    const elapsedTime = currTime / 1000 - this.startTime / 1000;
+
+    let easingFunc: (x: number) => number;
+    switch (this.easing) {
+      case 'linear':
+        easingFunc = linear;
+        break;
+      case 'easein':
+        easingFunc = easeIn;
+        break;
+      case 'flip':
+        easingFunc = flip;
+        break;
+      case 'easeout':
+        easingFunc = easeOut;
+        break;
+      default:
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-case-declarations
+        const __exhaustiveCase: never = this.easing;
+        throw new Error('Not all easing functions are handled');
```

Comment:
Default is already never in this case, so I think I'd recommend just doing something like this

```suggestion
      default:
        throw new Error(`Easing function ${this.easing} not handled!`);
```

<img width="1170" alt="Screenshot 2023-06-19 at 10 27 47 AM" src="https://github.com/jollyjerr/mindgraph/assets/51095001/ce859091-3c18-44c3-ae89-c54124fbbf15">

Reply:
@jollyjerr The `never` assignment provides compile type error checking for cases that aren't being handled (instead of runtime checking). In other words, if a new union was added to the `easing` type, the build will fail rather than a failure at runtime. 

See this: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking for a better explanation of what's going on. I added the `throw new Error` just for a "just in case this gets passed a build" scenario